File: go/chinese/src/xpyun-opensdk/service/httpclient.go
package service

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"xpyun-opensdk/model"
)

/**
 * 发送http的json请求
 *
 * @param url 请求url
 * @param jsonStr 发送的json字符串
 *
 */
func HttpPostJson(url string, data interface{}) *model.XPYunResp {
	b, _ := json.Marshal(data)

	fmt.Println(string(b))

	resp, err := http.Post(url, "application/json", bytes.NewBuffer(b))
	if err != nil {
		fmt.Println(err)
	}
	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println(fmt.Sprintf("post json error:%+v", err))
	}

	result := model.XPYunResp{
		HttpStatusCode: resp.StatusCode,
	}

	var content model.XPYunRespContent
	err = json.Unmarshal(body, &content)
	if err == nil {
		result.Content = &content
	} else {
		fmt.Println(fmt.Sprintf("unmarshal body failed, error:%+v", err))
	}

	return &result
}

File: python/chinese-demo/xpyunopensdk/service/httpclient.py
import requests
import json
import xpyunopensdk.model.model as model

"""
 发送http的json请求
 @param $url 请求url
 @param $jsonStr 发送的json字符串
 @return array
"""


def http_post_json(url, body):
    headers = {
        "Content-Type": "application/json; charset=UTF-8"
    }

    response = requests.post(url, data=json.dumps(body), headers=headers)

    result = model.XPYunResp
    result.httpStatusCode = response.status_code
    jsonObj = json.loads(response.text)
    result.content = model.XPYunRespContent
    result.content.code = jsonObj["code"]
    result.content.msg = jsonObj["msg"]
    # result.content.data = jsonObj["data"]
    result.content.serverExecutedTime = jsonObj["serverExecutedTime"]

    return result

